### PR TITLE
Socket implementation

### PR DESCRIPTION
**Pull request by Saptashrungi**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 
 app.get('/', function (req, res) {
-  res.send('<h1>Hello World!</h1>');
+  res.sendFile(__dirname + '/index.html');
 })
 
-app.listen(process.env.PORT || 3000,  function() {
+var io = require('./server/socket/index.js');
+io.attach(server)
+io.on('connection', function(socket){
```
Comment (kunaltawatia) [inline index.js:11]: This whole event function should have gone in the `server/socket/index.js`

File: index.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
 var app = express();
+var server = require('http').Server(app);
 
 app.get('/', function (req, res) {
-  res.send('<h1>Hello World!</h1>');
+  res.sendFile(__dirname + '/index.html');
 })
 
-app.listen(process.env.PORT || 3000,  function() {
+var io = require('./server/socket/index.js');
```
Comment (kunaltawatia) [inline index.js:9]: Import this at the start of the file, looks good there.
Also,
```js
var io = require('./server/socket');
```
would work.

File: server/socket/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+var io = require('socket.io')();
+
+module.exports = (io)
```
Comment (kunaltawatia) [inline server/socket/index.js:3]: No parentheses required.

**Comment by Saptashrungi** [comment]
In the recent commit there is a bug. Its not disconnecting. Client disconnected is not being displayed in the console.

**Comment by kunaltawatia** [comment]
Fixed in #26

**Pull request closed without merging**